pyomo/gdp/plugins/bigm.py

```diff
@@ -654,10 +654,10 @@ def _transform_constraint(self, obj, disjunct, bigMargs, arg_list,
 
             if c.lower is not None and M[0] is None:
                 M = (self._estimate_M(c.body, name)[0] - c.lower, M[1])
-                bigm_src[c] = M
+                bigm_src[c] = (None, M)
             if c.upper is not None and M[1] is None:
                 M = (M[0], self._estimate_M(c.body, name)[1] - c.upper)
-                bigm_src[c] = M
+                bigm_src[c] = (None, M)
```

Comment:
Will this end up "lying" about the source if a user sets a suffix with something like `(None, 10)` for an equality constraint?

Reply:
Yikes, you're right. There can be more than one source for a constraint... Should this return a list? Ick.

Reply:
@carldlaird: thoughts?  

What do we want to know here?  In general, is can be useful to know *where* the M value came from (and there could be more than one M value), and *what( the M value is.  The original idea was if you know where (e.g., from which suffix/dict, and what index/item), then the value is just an item lookup.  Carl advocated returning all three.

What about the following:
```python
(lower_M, lower_src, lower_idx), (upper_M, upper_src, upper_idx)
```
...or would it make more sense to keep the src, idx together:
```python
(lower_M, (lower_src, lower_idx)), (upper_M, (upper_src, upper_idx))
```
...but this makes a "triply-nested" structure - perhaps too complex?